eos/b-decays/b-to-gamma-l-nu.cc

```diff
@@ -0,0 +1,214 @@
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/b-decays/b-to-gamma-l-nu.hh>
+#include <eos/form-factors/mesonic.hh>
+#include <eos/maths/integrate.hh>
+#include <eos/maths/power-of.hh>
+#include <eos/models/model.hh>
+#include <eos/utils/options-impl.hh>
+#include <eos/utils/private_implementation_pattern-impl.hh>
+
+#include <tuple>
+
+namespace eos
+{
+    using std::norm;
+
+    /*
+     * Decay: B_u -> gamma l nubar, cf. [BBJW:2018A], [BR:2011A]
+     *
+     * Lepton and neutrino are assumed to be massless
+     */
+    template <>
+    struct Implementation<BToGammaLeptonNeutrino>
+    {
+        std::shared_ptr<Model> model;
+        std::shared_ptr<FormFactors<PToGamma>> form_factors;
+
+        UsedParameter alpha_qed;
+        UsedParameter g_fermi;
+        UsedParameter v_ub_abs;
+        UsedParameter hbar;
+
+        UsedParameter m_B;
+        UsedParameter f_B;
+        UsedParameter tau_B;
+
+        static const constexpr double e_l = -1.0;
+
+        static const std::vector<OptionSpecification> options;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            model(Model::make(o.get("model", "SM"), p, o)),
+            form_factors(FormFactorFactory<PToGamma>::create("B->gamma::" + o.get("form-factors", "FLvD2022QCDF"), p, o)),
+            alpha_qed(p["QED::alpha_e(m_b)"] ,u),
+            g_fermi(p["WET::G_Fermi"], u),
+            v_ub_abs(p["CKM::abs(V_ub)"], u),
+            hbar(p["QM::hbar"], u),
+            m_B(p["mass::B_u"], u),
+            f_B(p["decay-constant::B_u"], u),
+            tau_B(p["life_time::B_u"], u)
+        {
+            u.uses(*model);
+        }
+
+        double decay_width(const double & E_gamma_min) const
+        {
+            // cf. [BBJW:2018A], eq. (2.6)
+            const double prefactor = alpha_qed * power_of<2>(g_fermi * v_ub_abs) / (6.0 * power_of<2>(M_PI));
+
+            auto integrand = [&](const double & E_gamma) -> double {
+                const double F_V = form_factors->F_V(E_gamma);
+                const double F_A = form_factors->F_A(E_gamma);
+                return m_B * power_of<3>(E_gamma) * (1.0 - 2.0 * E_gamma / m_B)
+                    * ( power_of<2>(std::abs(F_V)) + power_of<2>( F_A + e_l * f_B / E_gamma ) );
+            };
+
+            return prefactor * integrate<GSL::QAGS>(integrand, E_gamma_min, m_B / 2.0);
+        }
+
+        double branching_ratio(const double & E_gamma_min) const
+        {
+            return decay_width(E_gamma_min) * tau_B / hbar;
+        }
+
+        double diff_decay_width(const double & E_gamma, const double & costheta) const
+        {
+            const double E_ell = (-2.0 * E_gamma * m_B + power_of<2>(m_B)) / (2.0 * (-E_gamma + costheta * E_gamma + m_B));
```

Comment:
Is this compatible with the definition in 2210.09832?

Reply:
Apparently not compatible, needs to be adjusted. See [2210.09832, Appendix D.1](https://arxiv.org/abs/2210.09832). We should use ``costheta`` corresponding to ``cos(theta_W)`` as defined there.